### DISCUSSION TITLE
Allow Restricting GitHub Project only to users inside that Project, forbidding Organization members

### DISCUSSION THREAD

**Discussion by markvrc** [community/community#21270] [Projects and Issues]
Hello,

I'd like to ask if it's possible to allow to create Projects board inside organization that wouldn't be visible to all participants of the organization. Is it possible?

**Comment by mariadhari6** [comment]
hello @borseno 
You can configure permissions for creating repositories in your organization.

You can choose whether members can create repositories in your organization. If you allow members to create repositories, you can choose whether members can create both public and private repositories or public repositories only. Organization owners can always create any type of repository. You can following thi steps:

- In the top right corner of GitHub.com, click your profile photo, then click Your organizations.
![Organization text](https://docs.github.com/assets/cb-10767/images/help/profile/your-organizations.png)
- Next to the organization, click Settings
![Settings text](https://docs.github.com/assets/cb-5720/images/help/organizations/settings-button.png)
- In the "Access" section of the sidebar, click     Member privileges.
- Under "Repository creation", select one or more options.
![Private](https://docs.github.com/assets/cb-20994/images/help/organizations/repo-creation-perms-radio-buttons-fpt.png)

hope it helps...

**Reply by markvrc** [reply]
Hi, I meant to ask about GitHub Projects, not repositories.

**Reply by mariadhari6** [reply]
> Hi, I meant to ask about GitHub Projects, not repositories.

sorry, maybe you can set project visibility to private, and invite the members you want
![Screenshot from 2022-07-26 00-04-53](https://user-images.githubusercontent.com/74663874/180835444-794e82d7-3b07-4929-965f-0aea36d93348.png)

![Screenshot from 2022-07-26 00-05-23](https://user-images.githubusercontent.com/74663874/180835496-248d0151-c2a9-4de3-a8fe-c5b211193612.png)

**Reply by markvrc** [reply]
actually I did it this way.
The problem lies in the following. If I create a project **within an organization**, then all people in organization will get access to that project, even if it is private. And even if I set access inheritance to none in the organization settings.

**Reply by lukewar** [reply]
@borseno on the last screenshot above there is `Base role` tile. If you change that one to `None`, only* people who have direct invite are going to be able to access that project:

<img width="307" alt="CleanShot 2022-07-26 at 13 09 06@2x" src="https://user-images.githubusercontent.com/7632110/180992561-e2ac0f82-2a0f-43af-a285-3065ae58cbed.png">

*Please bear in mind that organisation owners (admins) will still be able to access that project.

**Comment by markvrc** [comment]
I have found a bug. 
Somehow a person can't see public projects when they're added to my organization (and when they have no access inherited from being a member). But they can see private projects, but cannot read issues in those projects.

**Comment by markvrc** [comment]
<img width="1437" alt="image" src="https://user-images.githubusercontent.com/33373431/180971924-4fc0503a-386d-4adf-af8c-347b6058f605.png">
Here - user can see 2 private projects, when they have no access inherited.
But, they don't see a public project. 
Which, if you open  it from an anonymous tab - is Protagonist Pad. https://github.com/orgs/RedDuck-Software/projects?query=is%3Aopen&type=new 

but they don't see that.

Regarding the initial problem that I encountered. I think I will not give accesses to users by default, and I will only give them accesses to Repositories they get invited to via teams. This way users won't be able to see Project's tickets, and they will be shown as hidden items.

**Reply by lukewar** [reply]
@borseno I am curious what does this user see when she/he click on the `Projects` tab on the left. Currently selected is `Recently viewed`, if that user never visited that Public project that would explain why it is not shown on that list.
